src/main/java/space/testflight/FlywayExtension.java

```diff
@@ -122,6 +131,51 @@ public void afterAll(ExtensionContext context) throws Exception {
     }
   }
 
+  @Override
+  public void postProcessTestInstance(Object testInstance, ExtensionContext context) throws Exception {
+    Field connectionHolderField = locateConnectionHolderField(testInstance.getClass());
+    if (connectionHolderField != null) {
+      Store classStore = getClassStore(context);
+      Store globalStore = getGlobalStore(context, classStore.get(MIGRATION_TAG, String.class));
+
+      String jdbcUrl = globalStore.get(JDBC_URL, String.class);
+      String jdbcUsername = globalStore.get(JDBC_USERNAME, String.class);
+      String jdbcPassword = globalStore.get(JDBC_PASSWORD, String.class);
+
+      connectionHolderField.setAccessible(true);
+      connectionHolderField.set(testInstance, provideConnectionHolder(jdbcUrl, jdbcUsername, jdbcPassword));
+    }
+  }
+
+  private <T> Field locateConnectionHolderField(Class<T> clazz) throws NoSuchFieldException {
+    try {
+      return clazz.getDeclaredField(CONNECTION_HOLDER_FIELD_NAME);
+    } catch (NoSuchFieldException e) {
+      if (clazz.getSuperclass() == null) {
+        return null;
+      }
+      return locateConnectionHolderField(clazz.getSuperclass());
+    }
+  }
+
+  private ConnectionHolder provideConnectionHolder(String jdbcUrl, String jdbcUsername, String jdbcPassword) {
```

Comment:
Uhm, I guess projects that don't have dbrider in its classpath will fail, when you explicitely reference a DBRider class here.
I think, you have to return "Object" here.
Thus saying, the creation of the object will be somewhat tricky.

Reply:
I think, you should do something like
field.getType().getClassLoader().loadClass("com.github.database.rider.core.connection.ConnectionHolderImpl").newInstance(connection);